src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -3,145 +3,215 @@
-  const [selectedDataSourceId, setSelectedDataSourceId] = useState<string | undefined>(undefined);
-  const [searchText, setSearchText] = useState<string>('');
+  const [isDeleting, setIsDeleting] = React.useState<boolean>(false);
+  const [selectedDataSources, setSelectedDataSources] = useState<DataSourceTableItem[]>([]);
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<
+    Record<string, React.ReactNode>
+  >({});
   const [isLoading, setIsLoading] = useState<boolean>(false);
-  const history = useHistory();
 
-  const fetchDataSources = useCallback(() => {
-    if (featureFlagStatus && selectedDataSourceId === undefined) return;
+  /* Table selection handlers */
+  const onSelectionChange = (selected: DataSourceTableItem[]) => {
+    setSelectedDataSources(selected);
+  };
 
-    const endpoint =
-      featureFlagStatus && selectedDataSourceId !== undefined
-        ? `${DATACONNECTIONS_BASE}/dataSourceMDSId=${selectedDataSourceId}`
-        : `${DATACONNECTIONS_BASE}`;
+  const selection = {
+    onSelectionChange,
+  };
 
-    setIsLoading(true);
+  const setDefaultDataSource = async () => {
+    try {
+      for (const dataSource of selectedDataSources) {
+        if (getDefaultDataSourceId(uiSettings) === dataSource.id) {
+          await setFirstDataSourceAsDefault(savedObjects.client, uiSettings, true);
+        }
+      }
+    } catch (e) {
+      notifications.toasts.addDanger(
+        i18n.translate('dataSourcesManagement.directQueryTable.setDefaultDataSourceFailMsg', {
+          defaultMessage:
+            'Unable to find a default datasource. Please set a new default datasource.',
+        })
+      );
+    } finally {
+      setIsDeleting(false);
+    }
+  };
 
-    http
-      .get(endpoint)
-      .then((res: DirectQueryDatasourceDetails[]) => {
-        const dataConnections = res.map((dataConnection: DirectQueryDatasourceDetails) => ({
-          name: dataConnection.name,
-          connectionType: dataConnection.connector,
-          dsStatus: dataConnection.status,
-        }));
-        setData(dataConnections);
+  /* Delete selected data sources*/
+  const onClickDelete = () => {
+    setIsDeleting(true);
+
+    deleteMultipleDataSources(savedObjects.client, selectedDataSources)
+      .then(() => {
+        setSelectedDataSources([]);
+        // Fetch data sources
+        fetchDataSources();
+        setIsModalVisible(false);
+        // Check if default data source is deleted or not.
+        // if yes, then set the first existing datasource as default datasource.
+        setDefaultDataSource();
       })
-      .catch((err) => {
-        notifications.toasts.addDanger('Could not fetch data sources');
+      .catch(() => {
+        notifications.toasts.addDanger(
+          i18n.translate('dataSourcesManagement.directQueryTable.deleteDataSourceFailMsg', {
+            defaultMessage:
+              'Error occurred while deleting selected records for Data sources. Please try it again',
```

Comment:
This message seems off. How about:
```suggestion
              'An error occurred while attempting to delete the selected data sources. Please try again.',
```